python-lib/dku_io_utils.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+"""Module with read/write utility functions based on the Dataiku API"""
+
+import os
+from typing import Dict, AnyStr, List
+
+import pandas as pd
+
+import dataiku
+
+# ==============================================================================
+# CLASS AND FUNCTION DEFINITION
+# ==============================================================================
+
+
+def generate_path_df(folder: dataiku.Folder, file_extensions: List[AnyStr], path_column: AnyStr) -> pd.DataFrame:
```

Comment:
I am not a big fan of the idea of mocking. It's all too easy to create a fake mock object that will always pass the tests. This function does depend on the dataiku API, so I prefer to specify it clearly, so that I know it needs to be tested in the integration testing scenarios.

Reply:
Keeping for now as per our conversation - we may do a refactoring later on